urlaa914a1de2_xu_hao_erilex_tgz-pJ8-erilex_esl_ESLJ8/src/erilex/builder/RuleBuilder.java

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2009 Hao Xu
+ertranne@hotmail.com
+
+This file is part of EriLex.
+
+EriLex is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+EriLex is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with EriLex; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package erilex.builder;
+
+import erilex.*;
+import erilex.codegen.Type;
+
+/**
+ *
+ * @author ertri
+ */
+public class RuleBuilder implements Builder<Object> {
+
+    private final String name;
+    private  String env;
+    private  String type;
+    private final boolean aux;
+    private final boolean lex;
+    private final Grammar grammar;
+    private ExpressionBuilder<RuleBuilder, Object> exp;
+    private int arity;
+    private Type envType;
+    private Type typeType;
+
+    public RuleBuilder(String nameParam, Grammar g) {
+        name = nameParam;
+        aux = false;
+        lex = false;
+        grammar = g;
+    }
+
+    RuleBuilder(String nameParam, boolean auxParam, boolean lexParam, Grammar g) {
+        name = nameParam;
+        aux = auxParam;
+        lex = lexParam;
+        grammar = g;
+    }
+
+    public ExpressionBuilder<RuleBuilder, Object> start() {
+        return exp = new ExpressionBuilder(this, grammar);
+    }
+    public ExpressionBuilder<RuleBuilder, Object> start(int p) {
+        this.arity = p;
+        return exp = new ExpressionBuilder(this, grammar);
+    }
+    public ExpressionBuilder<RuleBuilder, Object> start(int p, String env, String type) {
+        this.arity = p;
+        this.env = env;
+        this.type = type;
+        return exp = new ExpressionBuilder(this, grammar);
+
+    }
+    public ExpressionBuilder<RuleBuilder, Object> start(int p, Type env, Type type) {
+        this.arity = p;
+        this.envType = env;
+        this.typeType = type;
+        return exp = new ExpressionBuilder(this, grammar);
+
+    }
+    public Production build() {
+        return null;
+    }
+
+    @Override
+    public Object exp(Production p) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.